commands/createfixpullrequests.go

```diff
@@ -95,113 +104,186 @@ func (cfp *CreateFixPullRequestsCmd) scan(scanSetup *utils.ScanDetails, currentW
 	return scanResults, isMultipleRoots, nil
 }
 
-func (cfp *CreateFixPullRequestsCmd) fixImpactedPackagesAndCreatePRs(project utils.Project, repoGitParams *utils.Git, branch string,
-	client vcsclient.VcsClient, scanResults []services.ScanResponse, currentWd string, isMultipleRoots bool) (err error) {
-	fixVersionsMap, err := cfp.createFixVersionsMap(&project, scanResults, isMultipleRoots)
+func (cfp *CreateFixPullRequestsCmd) fixImpactedPackagesAndCreatePRs(scanResults []services.ScanResponse, isMultipleRoots bool) (err error) {
+	fixVersionsMap, err := cfp.createFixVersionsMap(scanResults, isMultipleRoots)
 	if err != nil {
 		return err
 	}
+
 	// Nothing to fix, return
 	if len(fixVersionsMap) == 0 {
-		log.Info("Didn't find vulnerable dependencies with existing fix versions for", repoGitParams.RepoName)
+		log.Info("Didn't find vulnerable dependencies with existing fix versions for", cfp.details.RepoName)
 		return nil
 	}
+
 	log.Info("Found", len(fixVersionsMap), "vulnerable dependencies with fix versions")
+	return cfp.fixVulnerablePackages(fixVersionsMap)
+}
 
-	// Create temp working directory
-	wd, err := fileutils.CreateTempDir()
+func (cfp *CreateFixPullRequestsCmd) fixVulnerablePackages(fixVersionsMap map[string]*FixVersionInfo) (err error) {
+	cfp.gitManager, err = utils.NewGitManager(cfp.dryRun, cfp.dryRunRepoPath, ".", "origin", cfp.details.Token, cfp.details.Username)
+	if err != nil {
+		return err
+	}
+
+	clonedRepoDir, restoreBaseDir, err := cfp.cloneRepository()
 	if err != nil {
 		return err
 	}
 	defer func() {
-		e := fileutils.RemoveTempDir(wd)
-		if err == nil {
-			err = e
+		e1 := restoreBaseDir()
+		e2 := fileutils.RemoveTempDir(clonedRepoDir)
+		if err == nil && (e1 != nil || e2 != nil) {
+			err = fmt.Errorf("%s\n%s", e1, e2)
 		}
 	}()
-	log.Debug("Created temp working directory:", wd)
 
-	// Clone the content of the repo to the new working directory
-	gitManager, err := utils.NewGitManager(cfp.dryRun, cfp.dryRunRepoPath, ".", "origin", repoGitParams.Token, repoGitParams.Username)
+	// Fix all impacted packages
+	for impactedPackage, fixVersionInfo := range fixVersionsMap {
+		if err = cfp.fixSinglePackage(impactedPackage, fixVersionInfo); err != nil {
+			log.Warn(err)
+		}
+		// After finishing to work on the current vulnerability we go back to the base branch to start the next vulnerability fix
```

Comment:
...vulnerability we ... --> ...vulnerability, we ...

Reply:
done